app/src/main/java/io/gnosis/safe/ui/safe/settings/safe/AdvancedSafeSettingsViewModel.kt

```diff
@@ -0,0 +1,43 @@
+package io.gnosis.safe.ui.safe.settings.safe
+
+import io.gnosis.data.models.SafeInfo
+import io.gnosis.data.repositories.SafeRepository
+import io.gnosis.safe.ui.base.AppDispatchers
+import io.gnosis.safe.ui.base.BaseStateViewModel
+import pm.gnosis.model.Solidity
+import javax.inject.Inject
+
+class AdvancedSafeSettingsViewModel
+@Inject constructor(
+    private val safeRepository: SafeRepository,
+    appDispatchers: AppDispatchers
+) : BaseStateViewModel<AdvancedSafeSettingsState>(appDispatchers) {
+
+    override fun initialState(): AdvancedSafeSettingsState = AdvancedSafeSettingsState()
+
+    fun load() {
+        safeLaunch {
+            safeRepository.getActiveSafe()?.let { safe ->
+                val safeInfo = safeRepository.getSafeInfo(safe.address)
+                updateState {
+                    AdvancedSafeSettingsState(
+                        isLoading = false,
+                        viewAction = LoadSafeInfo(safeInfo)
+                    )
+                }
+            }
+        }
+    }
+
+    fun isDefaultFallbackHandler(fallbackHandle: Solidity.Address): Boolean =
```

Comment:
this should be part of the safe repository

Reply:
I figured, because it is only an equality check, that putting this into the safeRepository would be a bit overkill. When there are more fallback handlers address we need to observe, kind of like with master copy, I would have something more generic, but for now I think this ok and it's also covered with tests which is nice :)